src/runtime.rs

```diff
@@ -446,6 +439,8 @@ struct SharedState<DB: Database> {
     dependency_graph: Mutex<DependencyGraph<DB>>,
 }
 
+impl<DB> std::panic::RefUnwindSafe for SharedState<DB> where DB: Database {}
```

Comment:
This is missing `DB::DatabaseStorage: std::panic::RefUnwindSafe,`?

Reply:
Ah, yeah, you're right.

Reply:
I forgot that this was stored in the shared-state

Reply:
Should we have a compile-fail test for non refunwind safe data?

Or perhaps we should just bound keys & values by `RefUnwindSafe`? Doesn't "keys&values must not contain observable interior mutability" implies unwind safety?

Reply:
Debatable. We always clone the values that we give back to you etc. I suppose you could put "mutable shared state" in there -- it'd just be very, very wrong.

Reply:
So yeah this:

> Doesn't "keys&values must not contain observable interior mutability" implies unwind safety?

I think this is true